### PR TITLE
[TACHYON-1238] Update LOG message formats in tachyon-common to avoid string concatenation performance hit

### PR DESCRIPTION
**Pull request by sisirkoppaka**
Updated LOG messages in tachyon-common to avoid string concatenation performance hit.

### CODE REVIEW

**Comment by gpang** [comment]
ok to test

**Comment by gpang** [comment]
Jenkins, add to whitelist

File: common/src/main/java/tachyon/LeaderSelectorClient.java [hunk 1]
```diff
@@ -142,12 +142,12 @@ public void stateChanged(CuratorFramework client, ConnectionState newState) {
   public void takeLeadership(CuratorFramework client) throws Exception {
     mIsLeader.set(true);
     if (client.checkExists().forPath(mLeaderFolder + mName) != null) {
-      LOG.info("deleting zk path: " + mLeaderFolder + mName);
+      LOG.info("deleting zk path: {} {}", mLeaderFolder, mName);
```
Comment (gpang) [inline common/src/main/java/tachyon/LeaderSelectorClient.java]: The previous code had no space between the folder and name.

File: common/src/main/java/tachyon/LeaderSelectorClient.java [hunk 2]
```diff
@@ -142,12 +142,12 @@ public void stateChanged(CuratorFramework client, ConnectionState newState) {
...
       client.delete().forPath(mLeaderFolder + mName);
     }
-    LOG.info("creating zk path: " + mLeaderFolder + mName);
+    LOG.info("creating zk path: {} {}", mLeaderFolder, mName);
```
Comment (gpang) [inline common/src/main/java/tachyon/LeaderSelectorClient.java]: The previous code had no space between the folder and name.

File: common/src/main/java/tachyon/worker/WorkerClient.java [hunk 3]
```diff
@@ -219,7 +219,7 @@ private synchronized boolean connect() throws IOException {
       int port = mWorkerNetAddress.rpcPort;
       mWorkerAddress = new InetSocketAddress(host, port);
       mWorkerDataServerAddress = new InetSocketAddress(host, mWorkerNetAddress.dataPort);
-      LOG.info("Connecting " + (mIsLocal ? "local" : "remote") + " worker @ " + mWorkerAddress);
+      LOG.info("Connecting {} worker @ {}", (mIsLocal ? "local" : "remote"), mWorkerAddress);
```
Comment (gpang) [inline common/src/main/java/tachyon/worker/WorkerClient.java]: NIT: could make it "Connecting to ..."?

**Comment by gpang** [comment]
Thanks for the improvement! I left a few minor comments.

**Comment by sisirkoppaka** [comment]
@gpang Thanks for the comments! Updated with changes.

**Comment by gpang** [comment]
LGTM

**Comment by yupeng9** [comment]
LGTM. Thanks for the improvement.

**Pull request merged**